### PR TITLE
Feature/form api

### PR DESCRIPTION
**Pull request by borkarsaish65**

### CODE REVIEW

File: controllers/v1/formsController.js [hunk 1]
```diff
@@ -22,6 +22,219 @@ module.exports = class Forms extends Abstract {
     return 'forms';
   }
 
+  	/**
+	 * @api {post} /project/v1/forms/create
```
Comment (VISHNUDAS-tunerlabs) [inline controllers/v1/formsController.js:26]: @borkarsaish65 wrong url

File: controllers/v1/formsController.js [hunk 2]
```diff
@@ -22,6 +22,219 @@ module.exports = class Forms extends Abstract {
...
+	 * @apiVersion 1.0.0
+	 * @apiName create
+	 * @apiGroup Forms
+	 * @apiParamExample {json} Request-Body:
+	 * @apiHeader {String} X-authenticated-user-token Authenticity token
+	 * @apiSampleRequest /project/v1/forms/create
```
Comment (VISHNUDAS-tunerlabs) [inline controllers/v1/formsController.js:32]: @borkarsaish65 correctthe url

File: controllers/v1/formsController.js [hunk 3]
```diff
@@ -22,6 +22,219 @@ module.exports = class Forms extends Abstract {
...
+				return resolve({
+					status: error.status || httpStatusCode.internal_server_error.status,
+					message: error.message || httpStatusCode.internal_server_error.message,
+					errorObject: error,
+				})
+			}
+		})
+	}
+  	/**
+	 * @api {post} /project/v1/forms/update/:_id
```
Comment (VISHNUDAS-tunerlabs) [inline controllers/v1/formsController.js:97]: @borkarsaish65 Please correct the url

File: controllers/v1/formsController.js [hunk 4]
```diff
@@ -22,6 +22,219 @@ module.exports = class Forms extends Abstract {
...
+	 * @apiVersion 1.0.0
+	 * @apiName update
+	 * @apiGroup Forms
+	 * @apiParamExample {json} Request-Body:
+	 * @apiHeader {String} X-authenticated-user-token Authenticity token
+	 * @apiSampleRequest /project/v1/forms/update/663cc73584f1a0eb4e97e3db
```
Comment (VISHNUDAS-tunerlabs) [inline controllers/v1/formsController.js:103]: @borkarsaish65 Please correct the url

File: controllers/v1/formsController.js [hunk 5]
```diff
@@ -22,6 +22,219 @@ module.exports = class Forms extends Abstract {
...
+				return resolve({
+					status: error.status || httpStatusCode.internal_server_error.status,
+					message: error.message || httpStatusCode.internal_server_error.message,
+					errorObject: error,
+				})
+			}
+		})
+	}
+  	/**
+	 * @api {get} /project/v1/forms/read/:_id
```
Comment (VISHNUDAS-tunerlabs) [inline controllers/v1/formsController.js:166]: @borkarsaish65 Please correct the url

File: controllers/v1/formsController.js [hunk 6]
```diff
@@ -22,6 +22,219 @@ module.exports = class Forms extends Abstract {
...
+	 * @apiVersion 1.0.0
+	 * @apiName read
+	 * @apiGroup Forms
+	 * @apiParamExample {json} Request-Body:
+	 * @apiHeader {String} X-authenticated-user-token Authenticity token
+	 * @apiSampleRequest /project/v1/forms/read/663cc73584f1a0eb4e97e3db
```
Comment (VISHNUDAS-tunerlabs) [inline controllers/v1/formsController.js:172]: @borkarsaish65 Please correct the url

File: controllers/v1/formsController.js [hunk 7]
```diff
@@ -22,6 +22,219 @@ module.exports = class Forms extends Abstract {
...
+					const formData = await formsHelper.read(
+						req.params._id,
+						req.body,
+						req.userDetails.organizationId,
+						req.userDetails.userToken
+					)
+					return resolve(formData)
+				}
+			} catch (error) {
+        console.log(error)
```
Comment (VISHNUDAS-tunerlabs) [inline controllers/v1/formsController.js:229]: @borkarsaish65 remove unwanted consoles

File: controllers/v1/formsController.js [hunk 8]
```diff
@@ -22,6 +22,219 @@ module.exports = class Forms extends Abstract {
...
+				return resolve({
+					status: error.status || httpStatusCode.internal_server_error.status,
+					message: error.message || httpStatusCode.internal_server_error.message,
+					errorObject: error,
+				})
```
Comment (VISHNUDAS-tunerlabs) [inline controllers/v1/formsController.js:234]: @borkarsaish65 validator are missing for all the  apis mentioned in this controller. Please check and add

File: module/forms/helper.js [hunk 9]
```diff
@@ -10,7 +10,217 @@
...
+	 */
+	static getDefaultOrgId(userToken) {
+		return new Promise(async (resolve, reject) => {
+			try {
+				// call user-service to fetch default organization details
+				let defaultOrgDetails = await userService.fetchDefaultOrgDetails(
+					process.env.DEFAULT_ORGANISATION_CODE,
+					userToken
+				)
+				console.log(defaultOrgDetails,'defaultOrgDetails')
```
Comment (VISHNUDAS-tunerlabs) [inline module/forms/helper.js:34]: @borkarsaish65 remove consoles

File: generics/services/users.js [hunk 10]
```diff
@@ -70,7 +69,71 @@ const profile = function ( userId = "" ) {
...
+					'?organisation_code=' +
+					organisationIdentifier
+			}
+			const options = {
+				headers: {
+					'X-auth-token': 'bearer ' + userToken,
+					internal_access_token: process.env.INTERNAL_ACCESS_TOKEN,
+				},
+			}
+            console.log(url,'url')
```
Comment (VISHNUDAS-tunerlabs) [inline generics/services/users.js:102]: @borkarsaish65 please remove consoles before raising  PR

File: module/forms/helper.js [hunk 11]
```diff
@@ -10,7 +10,217 @@
...
+					filter = { _id: new ObjectId(_id), organizationId: orgId }
+				} else {
+					filter = { ...bodyData, organizationId: orgId }
+				}
+				const form = await formQueries.findOneForm(filter)
+				let defaultOrgForm
+				if (!form || !form._id) {
+					// call getDefaultOrgId() to get default organization details from user-service
+					const defaultOrgId = await this.getDefaultOrgId(userToken)
+					console.log(defaultOrgId,'<--defaultOrgId')
```
Comment (VISHNUDAS-tunerlabs) [inline module/forms/helper.js:159]: @borkarsaish65 remove consoles

**Review by VISHNUDAS-tunerlabs — CHANGES_REQUESTED** [review]
@borkarsaish65 reviewed on Oct 7

File: controllers/v1/formsController.js (hunk 1 above)
Comment (borkarsaish65) [reply controllers/v1/formsController.js:26]: done

File: databaseQueries/forms.js [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+module.exports = class Forms {
+	/**
+	 * form details.
+	 * @method
+	 * @name formDocuments
+	 * @param {Object} [filterData = "all"] - forms filter query.
+	 * @param {Array} [fieldsArray = "all"] - projected fields.
+	 * @param {Array} [skipFields = "none"] - field not to include
+	 * @returns {Array} form details.
+	 */
```
Comment (VISHNUDAS-tunerlabs) [inline databaseQueries/forms.js:17]: @borkarsaish65 Please add api configuration in the service if required

**Review by VISHNUDAS-tunerlabs — CHANGES_REQUESTED** [review]
Reiewed

File: constants/interface-routes/configs.json [hunk 13]
```diff
@@ -1156,6 +1156,58 @@
...
+            "inSequence": false,
+            "orchestrated": false,
+            "targetPackages": [
+                {
+                    "basePackageName": "survey",
+                    "packageName": "elevate-survey"
+                }
+            ]
+        },
+        {
```
Comment (VISHNUDAS-tunerlabs) [inline constants/interface-routes/configs.json:1186]: @borkarsaish65 update with /:id required right?

**Review by VISHNUDAS-tunerlabs — APPROVED** [review]

**Pull request merged**